pints/_nested/_rejection.py

```diff
@@ -0,0 +1,153 @@
+        """
+        Returns the number of active points that will be used in next run.
+        """
+        return self._acceptance_target
+
+    def iterations(self):
+        """
+        Returns the total number of iterations that will be performed in the
+        next run, including the non-adaptive and burn-in iterations.
+        """
+        return self._iterations
+
+    def run(self):
+        """See: :meth:`pints.MCMC.run()`."""
+        
+        # Report the current settings
+        if self._verbose:
+            print('Running nested rejection sampling')
+            print('Number of active points: ' + str(self._active_points))
+            print('Total number of iterations: ' + str(self._iterations))
+            print('Total number of posterior samples: ' + str(
+                self._posterior_samples))
+
+        # Problem dimension
+        d = self._dimension
+        
+        # go!
+        # generate initial random points by sampling from the prior
+        m_active = np.zeros((self._active_points, d + 1))
+        m_initial = self._prior.random_sample(self._active_points)
+        for i in range(0, self._active_points):
+            m_active[i, d] = self._log_likelihood(m_initial[i, :])
+        m_active[:, :-1] = m_initial
+        
+        # store all inactive points, along with their respective
+        # log-likelihoods (hence, d+1)
+        m_inactive = np.zeros((self._iterations, d + 1))
+        
+        # store weights
+        w = np.zeros(self._active_points + self._iterations)
+        
+        # store X values (defined in [1])
+        X = np.zeros(self._iterations + 1)
+        X[0] = 1
+        
+        # log marginal likelihood holder
+        v_log_Z = np.zeros(self._iterations + 1)
+        
+        # run iter
+        for i in range(0, self._iterations):
+            a_running_log_likelihood = np.min(m_active[:, d])
+            a_min_index = np.argmin(m_active[:, d])
+            X[i + 1] = np.exp(-(i + 1) / self._active_points)
+            w[i] = X[i] - X[i + 1]
+            v_log_Z[i] = a_running_log_likelihood
+            m_inactive[i, :] = m_active[a_min_index, :]
+            m_active[a_min_index, :] = reject_sample_prior(
+                a_running_log_likelihood, self._log_likelihood, self._prior)
+        
+        v_log_Z[self._iterations] = logsumexp(m_active[:, d])
+        w[self._iterations:] = float(X[self._iterations]) / float(
+            self._active_points)
+        m_samples_all = np.vstack((m_inactive, m_active))
+        log_Z = logsumexp(v_log_Z, b=w[0:(self._iterations + 1)])
+        # vLogP = m_samples_all[:, d] - log_Z + np.log(w)
+        vP = np.exp(m_samples_all[:, d] - log_Z) * w
+        m_theta = m_samples_all[:, :-1]
+        vIndex = np.random.choice(
+            range(0, self._iterations + self._active_points),
+            self._posterior_samples, p=vP)
+        m_posterior_samples = m_theta[vIndex, :]
+        return [m_posterior_samples, log_Z]
```

Comment:
This return type doesn't match the output specified in the interface for nested samplers.
(Return a tuple `(a, b)` is also more conventional than a list `[a, b])`